zkevm-circuits/src/evm_circuit/execution/error_oog_memory_copy.rs

```diff
@@ -165,17 +167,22 @@ impl<F: Field> ExecutionGadget<F> for ErrorOOGMemoryCopyGadget<F> {
         );
 
         let (is_warm, external_address) = if is_extcodecopy {
+            rws.next();
             (
-                block.get_rws(step, 1).tx_access_list_value_pair().0,
-                block.get_rws(step, 2).stack_value(),
+                rws.next().tx_access_list_value_pair().0,
+                rws.next().stack_value(),
             )
         } else {
             (false, U256::zero())
         };
 
-        let rw_offset = if is_extcodecopy { 3 } else { 0 };
-        let [dst_offset, src_offset, copy_size] = [rw_offset, rw_offset + 1, rw_offset + 2]
-            .map(|index| block.get_rws(step, index).stack_value());
+        if !is_extcodecopy {
+            rws.offset_reset();
+        };
```

Comment:
This line is not required.
If `is_extcodecopy` is false, then the three `rws.next()` in line 170~173 wouldn't have happened. rw_offset here would be still zero.

From the API perspective, rw_index represents the order or reads and writes. It should be monotonic increasing and we shouldn't design an API to make it decrease or reset to zero.
```suggestion
```